### DISCUSSION TITLE
Sending keypresses to a disabled webview2

### DISCUSSION THREAD

**Discussion by doskabouter** [MicrosoftEdge/WebView2Feedback#3939] [Q&A]
I'm trying to control a website through mouseclicks and sending keys to it, but that website should only get those events from my software, not from users clicking/typing into it.
So most of the time the webview is disabled, I only enable it (obviously) when I need to send a click or a key.

For mouseclicks I found a workable solution:

  webView.Enabled = true;
  webView.Focus();
  Application.DoEvents();
  Cursor.Position = new Point(x,y);
  CursorHelper.DoLeftMouseClick();
  webView.Parent.Focus();
  webView.Enabled = false;

however for sending keys that is not working, 
currently I have:

  webView.Enabled = true;
  Application.DoEvents();
  webView.Focus();
  Application.DoEvents();
  sendKeys("aaab");

and after that I'm stuck.
If I add a webView.Enabled=false after the sendkeys (or with an extra Application.DoEvents();) it's too early and the keys won't get sent to the webView.

Tried catching the webView_Key* events but they never get fired.
Also tried overriding webView's wndProc, but the messages passing through there don't give me any hint on when the keys are sent/processed by the webView

Any ideas on how to make this possible?

**Comment by ajtruckle** [comment]
Did you find a solution?

**Comment by doskabouter** [comment]
Unfortunately not...

**Comment by ajtruckle** [comment]
@doskabouter  See:

https://github.com/MicrosoftEdge/WebView2Feedback/issues/4483#issuecomment-2051847399

@champnic  We need `SendInputKeys` please or a viable workaround.
